### PR TITLE
Added support for specifying Kubelet CA Path

### PR DESCRIPTION
**Pull request by sabre1041**
Added support for specifying Kubelet CA Path

### CODE REVIEW

**Review by faisal-memon — APPROVED** [review]
Thanks @sabre1041

File: charts/spire/charts/spire-agent/README.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ A Helm chart to install the SPIRE agent.
...
 | `workloadAttestors.k8s.enabled`                     | Enables the Kubernetes workload attestor                                                                                                                                                          | `true`                                                                           |
 | `workloadAttestors.k8s.skipKubeletVerification`     | If true, kubelet certificate verification is skipped                                                                                                                                              | `true`                                                                           |
+| `workloadAttestors.k8s.kubeletCaPath`               | Path to the file containing CA certificates used to verify the kubelet certificate                                                                                                                | `/run/secrets/kubernetes.io/serviceaccount/ca.crt`                               |
```
Comment (kfox1111) [inline charts/spire/charts/spire-agent/README.md:91]: This doesn't follow the documented naming convention.
It should be kubeletCAPath

I'm not seeing where it is being mounted into the container too. Are we messing some code?

**Review by kfox1111 — COMMENTED** [review]
Thanks for the contribution.

A couple of issues

**Comment by kfox1111** [comment]
This is related to some of the discussion that was happening in https://github.com/spiffe/helm-charts-hardened/issues/56

specifically around, how do we find and mount the ca the kubelet is using into the spire agent. Most of the time, it is not the same CA used for the apiserver.

File: charts/spire/charts/spire-agent/README.md (hunk 1 above)
Comment (kfox1111) [reply charts/spire/charts/spire-agent/README.md:91]: Oh, its just assuming that its already there, and its defaulting to the apiserver's CA. In vanilla Kubernetes, that is not the case. :/

**Comment by kfox1111** [comment]
Oh. Found the issue I was looking for:
https://github.com/spiffe/helm-charts-hardened/issues/240

Has a lot of details around a proposed solution

**Comment by kfox1111** [comment]
https://github.com/spiffe/spire/issues/5370 is related.

**Comment by kfox1111** [comment]
I'm thinking, we may not want to allow setting kubelet_ca_path directly at all.
I can think of a few different ways to set it (Types):

KubeletCertificate:
needs a path on disk to the kubelet certificate. On most clusters, its self signed. But it will be just fine to check the tcp connection against the cert kubelet is using itself to serve with.

APIServerSigned:
use the CA provided in /run/secrets/kubernetes.io/serviceaccount/ca.crt.

Specified:
  here's the ca to use as a blob of text

And maybe an 'Auto" or some kind, if the default for vanilla might be different then the default for openshift.

On openshift, are the nodes all self signed by default like in vanilla?

**Pull request merged**